src/library/vm/vm_float.cpp

```diff
@@ -109,10 +110,22 @@ void initialize_vm_float() {
     DECLARE_VM_BUILTIN(name({"native", "float", "sqrt"}),  [](vm_obj const & a) {return mk_vm_float(std::sqrt(to_float(a)));});
     DECLARE_VM_BUILTIN(name({"native", "float", "cbrt"}),  [](vm_obj const & a) {return mk_vm_float(std::cbrt(to_float(a)));});
     DECLARE_VM_BUILTIN(name({"native", "float", "abs"}),   [](vm_obj const & a) {return mk_vm_float(std::abs(to_float(a)));});
-    DECLARE_VM_BUILTIN(name({"native", "float", "ceil"}),  [](vm_obj const & a) {return mk_vm_int(std::ceil(to_float(a)));});
-    DECLARE_VM_BUILTIN(name({"native", "float", "floor"}), [](vm_obj const & a) {return mk_vm_int(std::floor(to_float(a)));});
-    DECLARE_VM_BUILTIN(name({"native", "float", "trunc"}), [](vm_obj const & a) {return mk_vm_int(std::trunc(to_float(a)));});
-    DECLARE_VM_BUILTIN(name({"native", "float", "round"}), [](vm_obj const & a) {return mk_vm_int(std::round(to_float(a)));});
+    DECLARE_VM_BUILTIN(name({"native", "float", "ceil"}),  [](vm_obj const & a) {
+        auto b = std::ceil(to_float(a));
+        if (!std::isfinite(b)) throw exception("native.float.ceil: result is not finite");
```

Comment:
```suggestion
        if (!std::isfinite(b)) return mk_vm_int(0);
```
Exceptions are very inconvenient in the VM, since there's no way to catch them.

Reply:
I based the decision off of
https://github.com/leanprover-community/lean/blob/b2ebf25256ada9ebf888bf5c5fe327e2022207b9/src/library/vm/vm_int.cpp#L197

But looking again that's a "if I tried to do what you asked I'd exhaust all your memory" situation.

I guess I should move this behavior to `mk_vm_int(double)` now that I don't need to specialize the error message?